src/scripting/lua_widget_attributes.cpp

```diff
@@ -60,36 +64,59 @@ static gui2::widget* find_child_by_index(gui2::widget& w, int i)
 				list->add_row(gui2::widget_item{});
 			}
 		}
-		return list->get_row_grid(i - 1);
+		luaW_pushwidget(L, *list->get_row_grid(i - 1));
+		return true;
 	} else if(gui2::multi_page* multi_page = dynamic_cast<gui2::multi_page*>(&w)) {
 		int n = multi_page->get_page_count();
 		if(i > n) {
 			for(; n < i; ++n) {
 				multi_page->add_page(gui2::widget_item{});
 			}
 		}
-		return &multi_page->page_grid(i - 1);
+		luaW_pushwidget(L, *&multi_page->page_grid(i - 1));
+		return true;
 	} else if(gui2::tree_view* tree_view = dynamic_cast<gui2::tree_view*>(&w)) {
 		gui2::tree_view_node& tvn = tree_view->get_root_node();
 		int n = tvn.count_children();
 		if(i > n) {
 			throw std::invalid_argument("out of range");
 		}
-		return &tvn.get_child_at(i - 1);
+		luaW_pushwidget(L, *&tvn.get_child_at(i - 1));
+		return true;
 	} else if(gui2::tree_view_node* tree_view_node = dynamic_cast<gui2::tree_view_node*>(&w)) {
 		int n = tree_view_node->count_children();
 		if(i > n) {
 			throw std::invalid_argument("out of range");
 		}
-		return &tree_view_node->get_child_at(i - 1);
+		luaW_pushwidget(L, *&tree_view_node->get_child_at(i - 1));
+		return true;
 	} else if(gui2::stacked_widget* stacked_widget = dynamic_cast<gui2::stacked_widget*>(&w)) {
 		int n = stacked_widget->get_layer_count();
 		if(i > n) {
 			throw std::invalid_argument("out of range");
 		}
-		return stacked_widget->get_layer_grid(i - 1);
+		luaW_pushwidget(L, *stacked_widget->get_layer_grid(i - 1));
+		return true;
+	} else if(gui2::menu_button* menu_button = dynamic_cast<gui2::menu_button*>(&w)) {
+		int n = menu_button->get_item_count();
+		if(i > n) {
+			throw std::invalid_argument("out of range");
+		}
+        using config_ptr = config*;
+        new(L, 0) config_ptr(menu_button->get_row(i - 1));
+		luaL_setmetatable(L, "menu_item");
```

Comment:
using a raw pointer here is unsafe since you cannot guarantee that the object it points to is still valid for a the lifetime of this lua object

Reply:
We were talking about that issue on IRC. I think the solution is to make it a `lua_ptr<menu_item>` instead, does that sound correct to you? (Where `menu_item` is `drop_down_menu::entry_data` renamed.)